src/Common/src/System/Globalization/IdnMapping.Windows.cs

```diff
@@ -20,14 +20,22 @@ private unsafe string GetAsciiCore(char* unicode, int count)
             }
 
             // Do the conversion
-            char[] output = new char[length];
-            length = Interop.mincore.IdnToAscii(flags, unicode, count, output, length);
-            if (length == 0)
+            const int StackAllocThreshold = 512; // arbitrary limit to switch from stack to heap allocation
+            if (length < StackAllocThreshold)
             {
-                ThrowForZeroLength("unicode", SR.Argument_IdnIllegalName, SR.Argument_InvalidCharSequenceNoIndex);
+                char* output = stackalloc char[length];
+                Interop.mincore.IdnToAscii(flags, unicode, count, output, length);
```

Comment:
I don't really understand why the length check is repeated:
Under what context would `length != 0` In the first check to IdnToAscii or IdnToUnciode and then equal 0 in the second check?

Strikes me as the only way the second call would fail with length 0 is if the output buffer was an incorrect length, which is guaranteed not to happen.